src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/PXGraphSemanticModel.cs

```diff
@@ -0,0 +1,195 @@
+                }
+                else
+                {
+                    implicitModel = new PXGraphSemanticModel(pxContext, d.GraphType, d.GraphTypeSymbol, cancellation);
+                    models.Add(implicitModel);
+                }
+
+                implicitModel.Initializers = implicitModel.Initializers.Add(info);
+            }
+
+            return models;
+        }
+
+        private static IEnumerable<InitDelegateInfo> GetInitDelegates(CancellationToken cancellation, PXContext pxContext, INamedTypeSymbol typeSymbol)
+        {
+            cancellation.ThrowIfCancellationRequested();
+
+            IEnumerable<SyntaxNode> declaringNodes = typeSymbol.DeclaringSyntaxReferences
+                                                     .Select(r => r.GetSyntax(cancellation));
+            InstanceCreatedEventsAddHandlerWalker walker = new InstanceCreatedEventsAddHandlerWalker(cancellation, pxContext);
+
+            foreach (SyntaxNode node in declaringNodes)
+            {
+                cancellation.ThrowIfCancellationRequested();
+                walker.Visit(node);
+            }
+
+            return walker.GraphInitDelegates;
+        }
+
+        private class InstanceCreatedEventsAddHandlerWalker : CSharpSyntaxWalker
+        {
+            private readonly CancellationToken _cancellation;
+            private readonly PXContext _pxContext;
+
+            public List<InitDelegateInfo> GraphInitDelegates { get; private set; } = new List<InitDelegateInfo>();
+
+            public InstanceCreatedEventsAddHandlerWalker(CancellationToken cancellation, PXContext pxContext)
+            {
+                _cancellation = cancellation;
+                _pxContext = pxContext;
+            }
+
+            public override void VisitInvocationExpression(InvocationExpressionSyntax node)
+            {
+                _cancellation.ThrowIfCancellationRequested();
+
+                SemanticModel semanticModel = _pxContext.Compilation.GetSemanticModel(node.SyntaxTree);
+
+                if (semanticModel.GetSymbolInfo(node, _cancellation).Symbol is IMethodSymbol symbol)
+                {
+                    bool isCreationDelegateAddition = _pxContext.PXGraphRelatedMethods.InstanceCreatedEventsAddHandler.Equals(symbol.ConstructedFrom);
+
+                    if (isCreationDelegateAddition)
+                    {
+                        INamedTypeSymbol graphSymbol = symbol.TypeArguments[0] as INamedTypeSymbol;
+                        ExpressionSyntax delegateNode = node.ArgumentList.Arguments.First().Expression;
+                        ISymbol delegateSymbol = semanticModel.GetSymbolInfo(delegateNode).Symbol;
+
+                        GraphInitDelegates.Add(new InitDelegateInfo(graphSymbol, delegateSymbol, delegateNode));
+                    }
+                }
+
+                base.VisitInvocationExpression(node);
+            }
+        }
+
+        private class InitDelegateInfo
```

Comment:
@metallmonster maybe this class could be changed into readonly struct

Reply:
@SENya1990 this would require to add three backing readonly field which will lead to more complex semantics of this type